File: Arrays and Hashing/valid-sudoku.js
/**
 * @param {character[][]} board
 * @return {boolean}
 */
var isValidSudoku = function (board) {
  let set = new Set();
  for (let i = 0; i < 9; i++) {
    for (let j = 0; j < 9; j++) {
      let element = board[i][j];
      if (element === ".") {
        continue;
      }
      let currentSize = set.size;
      set.add(`${board[i][j]} found at row ${i}`);
      set.add(`${board[i][j]} found at column ${j}`);
      set.add(
        `${board[i][j]} found at box ${Math.floor(i / 3)} ${Math.floor(j / 3)}`
      );
      if (currentSize !== set.size - 3) {
        return false;
      }
    }
  }
  return true;
};

File: Arrays and Hashing/valid-sudoku.py
class Solution:
    def isValidSudoku(self, board):
        hash_set = set()
        for r in range(9):
            for c in range(9):
                element = board[r][c]
                if element == '.':
                    continue
                current_size = len(hash_set)
                hash_set.add('add ' + element + 'to row ' + str(r))
                hash_set.add('add ' + element + 'to col ' + str(c))
                hash_set.add('add ' + element + 'to box ' +
                             str(r//3) + str(c//3))
                if current_size + 3 != len(hash_set):
                    return False
        return True
